### DISCUSSION TITLE
Apollo Server 3 PubSub Replacement?

### DISCUSSION THREAD

**Discussion by KeithGillette** [apollographql/apollo-server#5470] [Q&A] [answered]
The removal of integrated Subscription support in Apollo Server 3 also removed its integrated PubSub implementation. What lightweight, easy-to-deploy PubSub implementations are recommended for GraphQL Subscriptions? I don't see PubSub covered in [the Reenabling Subscriptions section of the AS3 Migration documentation](https://www.apollographql.com/docs/apollo-server/migration/#reenabling-subscriptions).

**Comment by KeithGillette** [comment, marked as answer]
Answering my own question, I see this is actually addressed in [the Subscriptions section of the Apollo Server 3 documentation](https://www.apollographql.com/docs/apollo-server/data/subscriptions/#the-pubsub-class), though not in the Migration section.

**Reply by glasser** [reply]
Glad you found this! PubSub is also mentioned in this part of the migration guide: https://www.apollographql.com/docs/apollo-server/migration/#exports-from-graphql-subscriptions

**Reply by KeithGillette** [reply]
Thanks! I expected it to be in the Subscriptions section and didn't scroll down far enough. Should've done a Find first.

**Reply by glasser** [reply]
Good call. https://github.com/apollographql/apollo-server/pull/5487
